examples/tokens/ERC1155ownable.vy

```diff
@@ -353,12 +361,34 @@ def uri(id: uint256) -> String[MAX_URI_LENGTH]:
     """
     return self._uri
 
+# URI #
+@external
+def setContractURI(contractUri: String[MAX_URI_LENGTH]):
+    """
+    @dev set the contractURI for the contract. points to collection metadata file
+    @param contractUri the new urcontractUri for the contract
+    """
+    assert not self.paused, "The contract has been paused"
+    assert self._contractUri != contractUri, "new and current URI are identical"
+    assert msg.sender == self.owner, "Only the contract owner can update the URI"
+    self._contractUri = contractUri
+    log evtContractURI(contractUri, 0)
+
+
+@view
+@external
+def contractURI() -> String[MAX_URI_LENGTH]:
```

Comment:
Could get around this by just have it be public

Reply:
Reworked it!